tests/models/mamba/test_modeling_mamba.py

```diff
@@ -447,10 +447,9 @@ def test_simple_generate(self, device):
 
         model = MambaForCausalLM.from_pretrained("state-spaces/mamba-130m-hf", torch_dtype=torch.float16)
         model.to(device)
-        model.config.use_cache = True
         input_ids = tokenizer("Hey how are you doing?", return_tensors="pt")["input_ids"].to(device)
 
-        out = model.generate(input_ids, do_sample=False, max_new_tokens=10)
+        out = model.generate(input_ids, do_sample=False, use_cache=True, max_new_tokens=10)
```

Comment:
it's true by default no?

Reply:
I removed `model.config.use_cache = True` just above this and moved it into the `generate`, to test that current changes are working and generate accepts "use_cache" in kwargs for Mamba. 

The real changes are the ones in `prepare_inputs_for_generation` which didn't pass `use_cache` in kwargs earlier